src/chains/celo/serializers.ts

```diff
@@ -206,32 +166,36 @@ export function assertTransactionCIP42(
       '`gasPrice` is not a valid CIP-42 Transaction attribute.',
     )
 
-  if (maxFeePerGas && maxFeePerGas > MAX_MAX_FEE_PER_GAS)
+  if (isPresent(maxFeePerGas) && maxFeePerGas > MAX_MAX_FEE_PER_GAS)
     throw new FeeCapTooHighError({ maxFeePerGas })
 
   if (
-    maxPriorityFeePerGas &&
-    maxFeePerGas &&
+    isPresent(maxPriorityFeePerGas) &&
+    isPresent(maxFeePerGas) &&
     maxPriorityFeePerGas > maxFeePerGas
   )
     throw new TipAboveFeeCapError({ maxFeePerGas, maxPriorityFeePerGas })
 
   if (
-    (gatewayFee && !gatewayFeeRecipient) ||
-    (gatewayFeeRecipient && !gatewayFee)
+    (isPresent(gatewayFee) && isEmpty(gatewayFeeRecipient)) ||
+    (isPresent(gatewayFeeRecipient) && isEmpty(gatewayFee))
   ) {
     throw new BaseError(
       '`gatewayFee` and `gatewayFeeRecipient` must be provided together.',
     )
   }
 
-  if (feeCurrency && !feeCurrency?.startsWith('0x')) {
+  if (isPresent(feeCurrency) && !isAddress(feeCurrency)) {
```

Comment:
This reads so much better with `isPresent` and `isAddress`, thanks for refactoring this!